exercise-testing-basics.js/test/students-submissions/yoav-ben-zvi.test.js

```diff
@@ -0,0 +1,176 @@
+
+		//Assert
+		expect(recievedResult).toEqual({
+			succeeded: true,
+			reasons: [],
+		});
+		// ✅ TASK: Create another test against the 'validateUser' method of the 'usersService':
+		//    Provide all the mandatory field and ensure that the response is satisfactory (as expected)
+	});
+
+	test("Testing the 'reasons' array against missing properties in the 'validateUser' method of 'usersService'", () => {
+		//Arrange
+		const UserExampleWithMissingProperties = {
+			name: "Kent",
+			address: "Moonlight road 181, Alaska",
+		};
+		const userServiceUnderTest = new usersService();
+
+		//Act
+		const recievedResult = userServiceUnderTest.validateUser(
+			UserExampleWithMissingProperties
+		);
+
+		//Assert
+		expect(recievedResult.reasons).toIncludeAllMembers(["no-name"]);
+		// ✅ TASK: Create another test against the 'validateUser' method of the 'usersService':
+		//    Don't pass at least two mandatory properties, and ensure that the response.succeeded is false
+		//    but also that the 'reasons' array has ALL the right reasons inside
+	});
+
+	test("'validateUser' method shoul throw an error when no arguments are passed", () => {
+		//Arrange
+		const userServiceUnderTest = new usersService();
+
+		//Act
+		// QUESTION: is there an act stage in this case?
+
+		//Assert
+		expect(() => userServiceUnderTest.validateUser()).toThrow(
+			"No user was provided"
+		);
+		// ✅ TASK: Create another test against the 'validateUser' method of the 'usersService':
+		//    Don't pass anything to the function and ensure it throws back an exception
+	});
+});
+
+describe("Testing the 'addUser' method of 'userService'", () => {
+	test("'addUser' method should return true when given valid input", async () => {
+		//Arrange
+		const userServiceUnderTest = new usersService();
+		const fullUserExample = {
+			name: "Kent",
+			familyName: "Beck",
+			zipCode: "32486-01",
+			address: "Moonlight road 181, Alaska",
+		};
+
+		//Act
+		const recievedResult = await userServiceUnderTest.addNewUser(
+			fullUserExample
+		);
+
+		//Assert
+		expect(recievedResult).toBe(true);
+	});
+	// ✅ TASK: Create the first test against the 'addUser' method of the 'usersService':
+	//    Pass a valid input and expect a positive response back
+});
+
+describe("Testing the 'deleteUser' method of the 'usersService'", () => {
+	test("'deleteUser' method should return a positive response when user is deleted", async (done) => {
+		//Arrange
+		const userServiceUnderTest = new usersService();
+		const fullUserExample = {
+			name: "Kent",
+			familyName: "Beck",
+			zipCode: "32486-01",
+			address: "Moonlight road 181, Alaska",
+		};
+		await userServiceUnderTest.addNewUser(fullUserExample);
+
+		//Act
+		userServiceUnderTest.deleteUser("Kent", (param1, param2) => {
+			//Assert
+			expect(param2.succeed).toBe(true);
+			done();
+		});
+
+		//I have to say - it took me a while to understand what is going on here, and going through
```

Comment:
@Yoavbenzvi First, this specific task mistakenely sneaked to this exercise, it should not have been here as it belongs to future lessons.

Is your comment and feel related to this task only (delete user) or to all tasks in this file?

Reply:
this comment was just for this specific test, other than that everything was quite clear. i'm a bit relieved to see that it wasn't something i was supposed to get completely by myself